### PR TITLE
Reservations: add billing fields to staff event required fields 

### PR DESCRIPTION
**Pull request by danjacob-anders**
https://andersinno.atlassian.net/jira/software/c/projects/TTVA/boards/11?modal=detail&selectedIssue=TTVA-114

Ensures that billing fields are included with other staff event fields for "normal" reservations.

### CODE REVIEW

File: resources/api/reservation.py [hunk 1]
```diff
@@ -134,29 +134,40 @@ def __init__(self, *args, **kwargs):
...
             request_user = self.context["request"].user
             is_staff_event = data.get("staff_event", False)
 
+            # staff events always have the same set of required fields
             if is_staff_event and resource.can_create_staff_event(request_user):
                 required = {"reserver_name", "event_description"}
 
+                # billing fields are also required if own use
+                if data.get("type", None) == Reservation.TYPE_NORMAL:
+                    required |= {
```
Comment (jopesy) [inline resources/api/reservation.py]: Just to make sure I got this right, these are only required if the fields are in the reservation metadata set? So for free resources that have a reservation form (metadata set) without these fields, they are also not required, right?

File: resources/api/reservation.py [hunk 2]
```diff
@@ -134,29 +134,40 @@ def __init__(self, *args, **kwargs):
...
+                        "billing_last_name",
+                        "billing_email_address",
+                    }
+            else:
+                required = resource.get_required_reservation_extra_field_names(
+                    cache=cache
+                )
+
+            for field_name in required:
+                if field_name in self.fields:
```
Comment (jopesy) [inline resources/api/reservation.py:162]: I guess this answer's the question above but wanted to double check that this doesn't make the billing fields required for resources that are free.

Comment (danjacob-anders) [reply resources/api/reservation.py:162]: Yes, that should be the case but it wouldn't hurt to add more tests to make sure.

Comment (danjacob-anders) [reply resources/api/reservation.py:162]: Thinking more about this, perhaps it should be sufficient to check these fields are present in the required metadata i.e.  

```python

            required = resource.get_required_reservation_extra_field_names(cache=cache)

            if is_staff_event and resource.can_create_staff_event(request_user):
                required_for_staff = {"reserver_name", "event_description"}

                # billing fields are also required if own use
                if data.get("type", None) == Reservation.TYPE_NORMAL:
                    required_for_staff |= {
                        field_name
                        for field_name in (
                            "billing_first_name",
                            "billing_last_name",
                            "billing_email_address",
                        )
                        if field_name in required
                    }
                required = required_for_staff

```

Comment (jopesy) [reply resources/api/reservation.py:162]: Yeah, that should do it

**Review by jopesy — APPROVED** [review]
LGTM

**Pull request merged**